src/ripple/app/ledger/Ledger.cpp

```diff
@@ -680,10 +680,14 @@ loadLedgerHelper(std::string const& sqlSuffix)
         rangeCheckedCast<std::uint32_t>(ledgerSeq64.value_or (0));
 
     uint256 prevHash, accountHash, transHash;
-    ledgerHash.SetHexExact (sLedgerHash.value_or(""));
-    prevHash.SetHexExact (sPrevHash.value_or(""));
-    accountHash.SetHexExact (sAccountHash.value_or(""));
-    transHash.SetHexExact (sTransHash.value_or(""));
+    if (sLedgerHash)
```

Comment:
This relies on a design decision to have the default `base_uint` constructor zero-initialize. While we do this sort of thing all over the place, I'd rather we don't make a habit of it, but strive to reduce reliance on it.